enterprise/models.py

```diff
@@ -1066,7 +1066,7 @@ class EnterpriseCustomerUser(TimeStampedModel):
     user_fk = models.IntegerField(
         null=True,
         blank=False,
-        db_index=False,
+        db_index=True,
```

Comment:
with this change, we might also need to specify in the `Meta` class for this model the name of the index to use on this field: https://docs.djangoproject.com/en/5.1/ref/models/options/#django.db.models.Options.indexes
it would be something like
```python
    class Meta:
        indexes = [
            models.Index(fields=["user_fk"], name="idx_enterprise_user_fk"),
        ]
```